esp/services/ws_cloud/ws_cloudService.cpp

```diff
@@ -149,10 +203,39 @@ bool CWsCloudEx::onGetServices(IEspContext& context, IEspGetServicesRequest& req
     return true;
 }
 
+// Indented for reference and ease of reading
+// constexpr const char* jsonpath = R"!!(
+//     [{range .items[*]}{"{"}
+//     "name": "{.metadata.name}",
+//     "creationTimestamp": "{.metadata.creationTimestamp}",
+//     "containerStatuses": [
+//       {range .status.containerStatuses[*]}{"{"}
+//         "ready": {.ready},
+//         "restartCount": {.restartCount},
+//         "name": "{.name}"
+//       {"}"},{end}
+//     ],
+//     "ports": [
+//       {range .spec.containers[*].ports[*]}{"{"}
+//         "containerPort": {.containerPort},
+//         "name": "{.name}",
+//         "protocol": "{.protocol}"
+//       {"}"},{end}
+//     ],
+//     "status": "{.status.phase}",
+//     {"}"},{end}]
+//     )!!";
+
+// Single line for compact logging
+constexpr const char* jsonpath = R"!!([{range .items[*]}{"{"}"name": "{.metadata.name}","creationTimestamp": "{.metadata.creationTimestamp}","containerStatuses": [{range .status.containerStatuses[*]}{"{"}"ready": {.ready},"restartCount": {.restartCount},"name": "{.name}"{"}"},{end}],"ports": [{range .spec.containers[*].ports[*]}{"{"}"containerPort": {.containerPort},"name": "{.name}","protocol": "{.protocol}"{"}"},{end}],"status": "{.status.phase}",{"}"},{end}])!!";
+
 void CK8sResourcesInfoCache::read()
 {
     StringBuffer podsBuf, servicesBuf;
-    readToBuffer("kubectl get pods --output=json", podsBuf);
+    // Due to how IPipeProcess parses parameters we need the jsonpath argument
+    // quoted as shown below, or it will be treated as multiple arguments.
+    VStringBuffer command("kubectl get pods --selector=app.kubernetes.io/part-of=HPCC-Platform -o 'jsonpath=%s'", jsonpath);
```

Comment:
minor: not sure its worthwhile keeping them as two diff strings?

Reply:
Do you mean to combine the `command` string and the `jsonpath` into one?